lib/src/folding/parsers/indent.dart

```diff
@@ -5,60 +5,98 @@ import '../foldable_block.dart';
 import '../foldable_block_type.dart';
 import 'abstract.dart';
 
-/// A parser for foldable blocks from lines indentation
+/// A parser for foldable blocks from lines indentation.
 class IndentFoldableBlockParser extends AbstractFoldableBlockParser {
-  final _openBlocksLinesByIndent = <int, int>{};
+  Map<int, int> _openBlocksLinesByIndent = <int, int>{};
+  List<int?> _linesIndents = [];
 
   @override
   void parse({
     Result? highlighted,
     Set<Object?> serviceCommentsSources = const {},
     required List<CodeLine> lines,
   }) {
-    _addWhitespacesBlocks(lines);
+    _parse(lines);
     finalize();
   }
 
-  void _addWhitespacesBlocks(List<CodeLine> lines) {
-    final lineIndents = _calculateLineIndents(lines);
+  @override
+  void finalize() {
+    super.finalize();
+    _openBlocksLinesByIndent = {};
```

Comment:
Can we do this in the end of `_parse()`? This will leave absolutely no chance of misuse while here there's still a chance a client will not call `finalize()`.